fetch.php

```diff
@@ -1,30 +1,76 @@
 <?php
 namespace Openpublishing\Fetch;
 
+// When authorization request fails retry 1 time
+$GLOBALS['openpublishing_auth_retry'] = true;
+
+function openpublishing_print_debug($msg) {
+    print('<span class="OP_debug" style="display:none;">' . $msg . '</span><br>');
+}
+
+function openpublishing_get_auth_token() {
+    $HOST = 'https://' . get_option('openpublishing_api_host') . '/auth/auth?';
+    $BRAND_ID = get_option('openpublishing_brand_id');
+    $url = $HOST . 'realm_id=' . $BRAND_ID . '&type=world';
+
+    $response = wp_remote_get($url);
+    $status = wp_remote_retrieve_response_code($response);
+
+    if (is_wp_error($response) ) {
+        openpublishing_print_debug($response->get_error_message());
+    }
+    if (200 == $status) {
+        $json = json_decode(wp_remote_retrieve_body($response));
+        update_option(openpublishing_auth_token, $json->{'auth_token'});
+    }
+}
+
+function openpublishing_get_with_auth($url) {
+    if (!get_option('openpublishing_auth_token')) {
+        openpublishing_get_auth_token();
+    }
+    // TODO: change sslverify when going live
+    $options = array( 'headers' => array( 'Authorization' => 'Bearer ' . get_option('openpublishing_auth_token') ), 'sslverify' => false);
+    $response = wp_remote_get($url, $options);
+    $status = wp_remote_retrieve_response_code($response);
+
+    if (is_wp_error($response)) {
+        openpublishing_print_debug($response->get_error_message());
+    }
+    elseif (200 == $status) {
+        $GLOBALS['openpublishing_auth_retry'] = true;
+        return $response;
+    }
+    else {
+        $response_message = wp_remote_retrieve_response_message($response);
```

Comment:
Please use guard clause (i.e. filp error handling with success block).

Wouldnt it make more sense to move the internal get to a new function `openpublishing_get_with_auth_impl` and implement the retry block on the main level?

```
function openpublishing_get_with_auth($url) {

for max_retries:
      if openpublishing_get_with_auth_impl()
           return ...
      refresh_bearer_token()
      sleep 1 
}

function openpublishing_get_with_auth_impl($url) {
    if error:
        return false
    return true
}
```

We would then no longer need the global variable.

Reply:
resolved partially